nucleus-api/src/main/java/io/github/nucleuspowered/nucleus/api/teleport/event/NucleusTeleportEvent.java

```diff
@@ -8,45 +8,94 @@
 import io.github.nucleuspowered.nucleus.api.util.MightOccurAsync;
 import org.spongepowered.api.entity.Transform;
 import org.spongepowered.api.entity.living.player.Player;
+import org.spongepowered.api.entity.living.player.User;
 import org.spongepowered.api.event.entity.MoveEntityEvent;
 import org.spongepowered.api.event.entity.living.humanoid.player.TargetPlayerEvent;
+import org.spongepowered.api.event.user.TargetUserEvent;
 import org.spongepowered.api.util.annotation.NonnullByDefault;
 import org.spongepowered.api.world.World;
 
+import java.util.UUID;
+
+import javax.annotation.Nullable;
+
 /**
  * These events are fired <em>before</em> teleportation actually happens.
  */
 @NonnullByDefault
-public interface NucleusTeleportEvent extends TargetPlayerEvent, CancelMessageEvent {
+public interface NucleusTeleportEvent extends CancelMessageEvent {
```

Comment:
No, this is a breaking change and v2 of the API is stable. This cannot be removed.